src/System.Text.Json/src/System/Text/Json/Serialization/JsonSnakeCaseNamingPolicy.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSnakeCaseNamingPolicy : JsonNamingPolicy
+    {
+        public override string ConvertName(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            const char underscore = '_';
+            const UnicodeCategory noneCategory = UnicodeCategory.Control;
```

Comment:
What does the "none" here mean?

Should also be "NoneCategory" rather than "noneCategory".

Reply:
It's a trick to make `previousCategory` non-nullable which is based on that any character from not mentioned in the method Unicode categories are replaced an underscore. Therefore, letters, decimals and control characters are preserved.